File: src/chipschallenge/clonebehaviors/CloneSameDirection.java

package chipschallenge.clonebehaviors;

import chipschallenge.Block;


public class CloneSameDirection extends CloneBehavior {

    private CloneSameDirection() {
    }
    private static CloneSameDirection mInstance = null;

    public static synchronized CloneSameDirection getInstance() {
        if (mInstance == null) {
            mInstance = new CloneSameDirection();
        }
        return mInstance;
    }

    public Block cloneIt(Block original) throws CloneNotSupportedException {
        return cloneTo(original, original.getFacing());
    }
}

File: ChipsChallenge.Shared/Clonebehaviors/CloneSameDirection.cs
namespace ChipsChallenge.Shared.Clonebehaviors
{
    public class CloneSameDirection : CloneBehavior
    {
        private static CloneSameDirection instance;

        private CloneSameDirection()
        {
        }

        public static CloneSameDirection Instance
        {
            get
            {
                lock (typeof(CloneSameDirection))
                {
                    return instance ?? (instance = new CloneSameDirection());
                }
            }
        }

        public override Block CloneIt(Block original)
        {
            return CloneTo(original, original.Facing);
        }
    }
}